src/mod/endpoints/mod_verto/mod_verto.c

```diff
@@ -1423,6 +1440,36 @@ static void detach_calls(jsock_t *jsock)
 	if (wake) attach_wake();
 }
 
+static switch_status_t cmd_profile(char **argv, int argc, switch_stream_handle_t *stream)
+{
+	const char *err;
+
+	if (argc < 2) {
+		stream->write_function(stream, "Invalid Args!\n");
+
+		return SWITCH_STATUS_SUCCESS;
+	}
+
+	if (argv[1] && !strcasecmp(argv[0], "restart") && !strcasecmp(argv[1], "all")) {
+		switch_xml_reload(&err);
+		stream->write_function(stream, "Restart all profiles\n");
+		/* hangup all calls */
+		drop_calls();
+		/* stop all profiles */
+		kill_profiles();
```

Comment:
New calls can come between drop calls and kill profiles?

Reply:
i guess it could happen.. there is no lock condition. 

What lock you think i should use?